lib/engine/game/g_18_royal_gorge/step/buy_train.rb

```diff
@@ -16,6 +16,25 @@ def room?(entity)
             # even when train tight, there's room for a self-rust
             return true if entity.trains.any? { |t| t.rusts_on == upcoming_train.name }
           end
+
+          def check_connected!(buyer, train)
+            return if @game.loading
+            return if train.owner == @game.depot
+
+            seller = train.corporation
+            seller_nodes = @game.graph.connected_nodes(seller).keys
+            buyer_nodes = @game.graph.connected_nodes(buyer).keys
+
+            common_nodes = seller_nodes & buyer_nodes
+            return if common_nodes.any? { |n| n.available_slots.positive? || n.tokened_by?(seller) || n.tokened_by?(buyer) }
```

Comment:
`available_slots` is only a method for city nodes. Maybe
```suggestion
            return if common_nodes.any? { |n| n.town? || n.available_slots.positive? || n.tokened_by?(seller) || n.tokened_by?(buyer) }
```